### PR TITLE
Trezor encrypt

### PR DESCRIPTION
**Pull request by petrkr**
###### Motivation for this change
Make new version of old PR trezorCryptKeyValue. This version uses latest official Trezord-go code base and it does not use 3rd party libraries for its work. As main it works with both Trezor one and Trezor T

###### Things done

Prepare LUKS environment
Maintainer
Added initial version of new trezorencrypt as replacement of trezorCryptKeyValue

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by petrkr** [comment]
Rebase to latest master. Updated version 0.1.1 with some fixes about exit codes

File: pkgs/tools/misc/trezorencrypt/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  name = "trezorencrypt-${version}";
+  version = "v0.1";
```
Comment (jb55) [inline pkgs/tools/misc/trezorencrypt/default.nix]: should be `version = "0.1"`

File: pkgs/tools/misc/trezorencrypt/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  # Fixes Cgo related build failures (see https://github.com/NixOS/nixpkgs/issues/25959 )
+  hardeningDisable = [ "fortify" ];
+
+  goPackagePath = "github.com/petrkr/trezorencrypt";
+
+  src = fetchFromGitHub {
+    owner  = "petrkr";
+    repo   = "trezorencrypt";
+    rev    = "${version}";
```
Comment (jb55) [inline pkgs/tools/misc/trezorencrypt/default.nix]: `rev = "v${version}"`

**Review by jb55 — CHANGES_REQUESTED** [review]
This looks neat, looking forward to trying it out. Just a few nits.

File: pkgs/tools/misc/trezorencrypt/default.nix (hunk 2 above)
Comment (petrkr) [reply pkgs/tools/misc/trezorencrypt/default.nix]: So name of nix's package will not contain that 'v' ?right?

Comment (jb55) [reply pkgs/tools/misc/trezorencrypt/default.nix]: correct, nixpkgs has version functions that expect the `0.1` form as well.

**Comment by jb55** [comment]
was 14443df793ee806187f51e87c2223ab6a706f5d9 meant to be included in this?

**Comment by jb55** [comment]
also please rebase+squash the last two commits into `trezorencrypt: init at 0.1.1`

**Comment by petrkr** [comment]
First i think to separate to another PR, i just before had whole PR and this is not breaking but it does not corespond with trezor encrypt package, i'll do separate then later

Squashing I can do, since initial version was not merged so have extra commit about change version is useless, i agree.

Regards

**Comment by petrkr** [comment]
Rebased to latest master, included notes from discuss.

**Review by jb55 — APPROVED** [review]
LGTM

**Comment by jb55** [comment]
I tested this with my trezor and it works great. I would merge but I don't have the ability to do so :(

**Comment by petrkr** [comment]
Found one bug in versiosn 0.1.1, so released 0.1.2. There is bug if you do not have set trezor's label. I will crash...

So now I rebased to latest master and squashed initial release to 0.1.2. You can test old and new as well.

**Pull request closed without merging**